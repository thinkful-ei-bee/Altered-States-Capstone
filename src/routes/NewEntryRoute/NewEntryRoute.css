/* testing, not important */
.cloudinary-thumb {
  width: 360px;
}

.new-entry-page {
  margin: auto;
  margin-top: 70px;
  max-width: 360px;
  margin-top: 90px;
}

.ne-title {
  text-align: left;
  padding: 0 5%;
}

.ne-title-top {
  padding-bottom: 10px;
}

.ne-title h3 {
  font-weight: 300;
  font-size: 16px;
}

.ne-title h2, .ne-title h3 {
  margin: 0;
}

.ta {
  position: relative;
  top: 10px;
}

.cloud-button {
  margin-top: 20px;
}

#parallax {
  min-height: 410px; 
  background-attachment: fixed;
  background-position-y: 38%;
  background-position-x: center;
  background-repeat: no-repeat;
  background-size: initial;
  width: 90%;
  margin: auto;
  margin-bottom: 3%;
}

.entry_form {
  display: inline-block;
  margin: auto;
  width: 100%;
  max-width: 360px;
  transition: 300ms;
}

.entry_form button {
  margin: auto;
  color: white;
  background-color: rgb(189, 62, 30);
  font-size: 16px;
  padding: 12px;
  width: 67%;
  float: right;
  border: none;
}

.entry_form a {
  margin: auto;
  color: rgb(65, 65, 65);
  background-color: white;
  font-size: 16px;
  width: 33%;
  float: left;
  padding: 12px 0;
  padding-bottom: 0;
}

.entry_form footer {
  position: fixed;
  width: 324px;
  max-width: 334px;
  left: 0;
  bottom: 0;
  right: 0;
  display: inline-block;
  margin: auto;
  padding-bottom: 0;
}

textarea {
  resize: none;
  display: block;
  margin: 20px auto;
  margin-bottom: 70px;
  width: 302px;
  height: 345px;
  padding: 10px;
  border: 1px solid darkgray;
  font-size: 16px;
}

.upload-button {
  border: 1px solid darkgray;
  background-color: white;
  height: 120px;
  width: 326px;
  transition: 300ms;
  margin: 18px;
  margin-top: 7px;
  margin-bottom: 28px;
}

.upload-button:hover {
  background-color: rgb(174, 212, 235);
}

@media only screen and (min-height: 700px) {
  textarea {
    height: 412px;
  }
}

@media only screen and (min-width: 768px) {
  .new-entry-page {
    display: grid;
    grid-template-columns: 10% 40% 40% 10%;
    grid-template-rows: 100px 30% 1fr;
    max-width: 1000px;
    margin-top: 100px;
  }
  .ne-title {
    text-align: center;
  }
  .ne-title-top {
    grid-column: 2/4;
    grid-row: 1/2;
  }
  .entry_form {
    grid-column: 2/3;
    grid-row: 2/3;
  }
  .entry_form footer {
    position: relative;
    grid-column: 3/4;
    grid-row: 3/4;
    margin-top: 10px;
  }
  textarea {
    height: 260px;
    width: inherit;
    margin-bottom: 0;
  }
  .cloud-button {
    grid-column: 3/4;
    grid-row: 2/3;
  }
  .cloud-button h3 {
    position: relative;
    bottom: 10px;
  }
  .upload-button {
    margin-top: 0px;
    height: 180px;
    width: 150px;
  }
  #parallax {
    grid-column: 3/4;
    grid-row: 2/3;
    margin: auto;
    margin-top: 20px;
    background-attachment: inherit;
    background-size: 50%;
    min-height: 285px;
    width: 318px;
  }
  .mood-container {
    grid-column: 3/4;
    grid-row: 3/4;
    height: 130px;
    margin-top: 85px;
  }


}