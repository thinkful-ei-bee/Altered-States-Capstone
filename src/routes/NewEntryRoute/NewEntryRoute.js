import React, { Component } from "react";
import { Redirect, Link } from 'react-router-dom';
import CloudinaryWidget from "../../components/CloudinaryWidget/CloudinaryWidget";
import EntryService from '../../services/entry-service'
import MoodSelector from '../../components/MoodSelector/moodSelector'
import './NewEntryRoute.css';


export default class NewEntryRoute extends Component {

  constructor(props) {
    super(props)
    this.state = {
      redirect: false,
      res_id: '',
      newEntry: {
        text: '',
        happiness: null,
        face_url: '',
        Anger: 0,
        Joy: 0,
        Fear: 0,
        Sadness: 0,
        Analytical: 0,
        Confident: 0,
        Tentative: 0,
        face_anger: 0,
        face_contempt: 0,
        face_disgust: 0,
        face_fear: 0,
        face_happiness: 0,
        face_neutral: 0,
        face_sadness: 0,
        face_surprise: 0
      }
    }
  }


  // Entries sent to Tone Analyzer need an 'Authorization' header with a base64 encoded 
  // username and password like so:
  // apikey:3489hgdvuh2384hfetc.etc.etc.etc.

  updateFaceUrl = async (url) => {
    await this.setState({newEntry: {...this.state.newEntry, face_url: url}}, () => console.log('face_url:', this.state.newEntry.face_url))
    document.getElementById('parallax').style.backgroundImage = `url(${this.state.newEntry.face_url})`
  }

  updateFaceData = (faceData) => {
    this.setState({newEntry: {...this.state.newEntry, ...faceData }}, () => console.log('face state:', this.state.newEntry))
  }

  updateEntry = (text) => {
    this.setState({ newEntry: {...this.state.newEntry, text }})
  }

  handleEntryTones = (tones) => {
    this.setState({newEntry: {...this.state.newEntry, ...tones }}, () => console.log('tones state:', this.state.newEntry))
  }

  handleSubmitEntry(event) {
    event.preventDefault()
    let tones = {}

    EntryService.postEntryToWatson(this.state.newEntry.text)
      .then(res => {
        let toneData = res.document_tone.tones

        for (let i in toneData) {
          tones[toneData[i].tone_name] = Math.floor(toneData[i].score * 50)
        }

        this.handleEntryTones(tones)
      })
      .then(() => {
        EntryService.postEntry(this.state.newEntry)
          .then(res => {
            this.setState({ res_id: res.id })
            this.setRedirect()
          })
      })
  }


  handleHappinessClick=(e)=>{ 

    if(!isNaN(e.target.value)){
      this.setState({
        newEntry: {...this.state.newEntry, happiness:Number(e.target.value)}
      },()=>{console.log('happiness state:', this.state.newEntry)} // should we send this to database from here or 
                                      //should we have one submit that will simply send all of State to database?
      )}
    }

  handleFinishedEntry(event, newEntry) {
    event.preventDefault();
    EntryService.postEntry(newEntry)
      .then(res => {
        this.setState({res_id: res.id})
        this.setRedirect()
      })
  }

  setRedirect() {
    this.setState({redirect: true})
  }

  renderRedirect() {
    if (this.state.redirect) {
      return <Redirect to={`/entry/${this.state.res_id}`} />
    }
  }

// <img src={this.state.newEntry.face_url} alt='uploaded selfie' className='cloudinary-thumb'/>
  handleDeleteSelfie() {
    EntryService.deleteSelfie(this.state.newEntry.face_url)
      .then(res => {
        this.setState({newEntry: {...this.state.newEntry, face_url: ''}}, () => console.log('state', this.state.newEntry))
        console.log('res', res)
      })
  }

  render() {
    return (
      <div>

        {this.state.newEntry.face_url && <div id='parallax'></div>}

        {!this.state.newEntry.face_url && 
          <CloudinaryWidget 
            updateFaceUrl={this.updateFaceUrl.bind(this)} 
            updateFaceData={this.updateFaceData.bind(this)} />}

        <MoodSelector handleClick={this.handleHappinessClick}/>

        <form className='entry_form' value={this.state.newEntry.text} onSubmit={(event) => this.handleSubmitEntry(event)}>
          <textarea 
            className='entry_area'
            maxLength='5000'
            onChange={(event) => this.updateEntry(event.target.value)}></textarea>
          <Link to='/'>CANCEL</Link>
          <button type='submit'>SUBMIT</button>
        </form>

        {this.renderRedirect()}
      </div>
    );
  }
}