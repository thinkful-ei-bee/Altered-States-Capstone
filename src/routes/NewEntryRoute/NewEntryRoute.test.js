import React from 'react';
import NewEntryRoute from './NewEntryRoute';
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() })

describe('DashboardRoute', () => {

  it('renders the complete page', () => {
    const wrapper = shallow(<NewEntryRoute />);
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})