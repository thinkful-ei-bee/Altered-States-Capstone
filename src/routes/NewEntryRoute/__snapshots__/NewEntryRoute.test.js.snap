// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardRoute renders the complete page 1`] = `
<div
  className="new-entry-page"
>
  <div
    className="ne-title ne-title-top"
  >
    <h2>
      Sat Jun 22
    </h2>
    <h3>
      New Entry
    </h3>
  </div>
  <CloudinaryWidget
    updateFaceData={[Function]}
    updateFaceUrl={[Function]}
  />
  <moodSelector
    handleClick={[Function]}
  />
  <form
    className="entry_form left"
    onSubmit={[Function]}
    value=""
  >
    <div
      className="ne-title ta"
    >
      <h3>
        How's it going? What's on your mind?
      </h3>
    </div>
    <textarea
      className="entry_area"
      maxLength="5000"
      onChange={[Function]}
    />
    <footer>
      <Link
        to="/"
      >
        CANCEL
      </Link>
      <button
        className="ne-submit"
        disabled={true}
        id="ne-submit"
        type="submit"
      >
        SUBMIT
      </button>
    </footer>
  </form>
</div>
`;
